file-check/action.yaml

```
@@ -0,0 +1,54 @@
+name: file-check
+description: ""
+
+inputs:
+  files:
+    description: ""
+    required: true
+  condition:
+    description: ""
+    required: true
+outputs:
+  exists:
+    description: ""
+    value: ${{ steps.file-check.outputs.exists }}
+
+runs:
+  using: composite
+  steps:
+    - name: File check
+      id: file-check
+      run: |
+        set -e
+
+        # Split files into an array
+        input_files="${{ inputs.files }}"
+        file_array=(${input_files})
+
+        # Loop through files
+        exist_files=()
+        for FILE in "${file_array[@]}"; do
+          if [[ -f $FILE ]]; then
+            echo "File $FILE exists"
+            exist_files+=("$FILE")
+          else
+            echo "File $FILE does not exist"
+          fi
+        done
+
+        if [[ ${{ inputs.condition }} = "or" ]]; then
+          if [[ ${#exist_files[@]} -gt 0 ]]; then
```

```suggestion
          if [[ "$num_found_files" -gt 0 ]]; then
```

And define `$num_found_files` somewhere.

---

(BTW I think this is optional. I forgot to write it.)

---

2dcc393

---

> (BTW I think this is optional. I forgot to write it.)

Thanks for additional comment. I have decided to accept your suggestion as it is more readable.

---

Oh, cool code! Nice work. :+1: